### PR TITLE
feat: add POST to create token and token as optional parameters on balance and simple-send-tx

### PR DESCRIPTION
**Pull request by pedroferreira1**
There are some features/fixes on this PR:

1. If I receive a `/start` to a key I already have, I stop the old wallet and start a new one. This fixes a bug when sending `/start` many times to the same key/wallet
2. Add optional token parameter  to `/balance`, `/simple-send/tx` and `/send-tx` endpoints
3. Add endpoint to create, mint and melt tokens (`/create-token`, `/mint-tokens`, `/melt-tokens`)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,6 +80,16 @@ app.post('/start', (req, res) => {
...
+  const walletID = req.body['wallet-id'];
+
+  if (walletID in wallets) {
+    // We already have a wallet for this key
+    // so we stop the old wallet and start the new one
+    const oldWallet = wallets[walletID];
+    oldWallet.stop();
+
+    delete wallets[walletID];
+  }
```
Comment (obiyankenobi) [inline index.js:92]: Why do it do it like this? Can't we just return saying this wallet is already started?

File: index.js [hunk 2]
```diff
@@ -124,6 +134,8 @@ walletRouter.use((req, res, next) => {
     return;
   }
 
+  console.log('Received request to', req.originalUrl)
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply index.js:137]: The idea here was to leave this log so we can see the requests when running the API

[the remark this answers is not in the document]
Comment (obiyankenobi) [reply index.js:137]: Got it, fine by me.

File: index.js (hunk 1 above)
Comment (obiyankenobi) [reply index.js:92]: Ah, didn't think about this other use. Actually, I think we should then return an error if he tries to start a wallet with an id already in use. He should then explicitly stop it and start another one.

Comment (pedroferreira1) [reply index.js:95]: Yeah, I agree with you. Changed the code to have this behaviour.

File: index.js [hunk 3]
```diff
@@ -80,6 +80,19 @@ app.post('/start', (req, res) => {
   if (req.body.passphrase) {
     walletConfig['passphrase'] = req.body.passphrase;
   }
+  const walletID = req.body['wallet-id'];
+
+  if (walletID in wallets) {
+    // We already have a wallet for this key
+    // so we log that it won't start a new one because
+    // it must first stop the old wallet and then start the new
+    console.log('Error starting wallet because this wallet-id is already used. You must stop the wallet using it first.');
```
Comment (msbrogli) [inline index.js:89]: Typo: `... wallet-id is already in use. You must stop the wallet first.`

File: index.js [hunk 4]
```diff
@@ -185,7 +202,9 @@ walletRouter.post('/simple-send-tx', (req, res) => {
   const wallet = req.wallet;
   const address = req.body.address;
   const value = parseInt(req.body.value);
-  const ret = wallet.sendTransaction(address, value);
+  // Expects object with {'uid', 'name', 'symbol'}
+  const token = req.body.token || null;
```
Comment (msbrogli) [inline index.js:206]: Use `tokenId` instead of `token`?

File: index.js [hunk 5]
```diff
@@ -201,7 +220,60 @@ walletRouter.post('/send-tx', (req, res) => {
   const wallet = req.wallet;
   const outputs = req.body.outputs;
   const inputs = req.body.inputs || [];
-  const ret = wallet.sendManyOutputsTransaction(outputs, inputs)
+  // Expects object with {'uid', 'name', 'symbol'}
+  const token = req.body.token || null;
```
Comment (msbrogli) [inline index.js:224]: Same here. `tokenId` instead of `token`.

File: index.js [hunk 6]
```diff
@@ -201,7 +220,60 @@ walletRouter.post('/send-tx', (req, res) => {
...
+      res.send(response);
+    }, (error) => {
+      res.send({success: false, error});
+    });
+  } else {
+    res.send({success: false, error: ret.message});
+  }
+});
+
+walletRouter.post('/create-token', (req, res) => {
```
Comment (msbrogli) [inline index.js:237]: Missing docstring.

File: index.js [hunk 7]
```diff
@@ -201,7 +220,60 @@ walletRouter.post('/send-tx', (req, res) => {
...
+      res.send(response);
+    }, (error) => {
+      res.send({success: false, error});
+    });
+  } else {
+    res.send({success: false, error: ret.message});
+  }
+});
+
+walletRouter.post('/mint-tokens', (req, res) => {
```
Comment (msbrogli) [inline index.js:255]: Missing docstring.

File: index.js [hunk 8]
```diff
@@ -201,7 +220,60 @@ walletRouter.post('/send-tx', (req, res) => {
...
+      res.send(response);
+    }, (error) => {
+      res.send({success: false, error});
+    });
+  } else {
+    res.send({success: false, error: ret.message});
+  }
+});
+
+walletRouter.post('/melt-tokens', (req, res) => {
```
Comment (msbrogli) [inline index.js:272]: Missing docstring.

**Review by msbrogli — APPROVED** [review]

File: index.js (hunk 4 above)
Comment (pedroferreira1) [reply index.js:206]: I agree 100% with you, however we should have a lib refactor to support this. 

I had this discussion with @obiyankenobi here: https://github.com/HathorNetwork/hathor-wallet-lib/pull/111#discussion_r544531482

File: index.js (hunk 3 above)
Comment (pedroferreira1) [reply index.js:89]: Fixed, thanks

File: index.js (hunk 6 above)
Comment (pedroferreira1) [reply index.js:238]: We don't really have any default docstring for APIs, should we start adding docstring to each endpoint?
What we have here is the `api-docs.js` which was not updated and I've just added all new endpoints and updated old ones. What do you think? Should we have the api-docs file and the docstring? Or a simple docstring version?

File: index.js (hunk 4 above)
Comment (msbrogli) [reply index.js:206]: Ok. Do we have another issue to refactor it later? A simple way is to create a new method that receives only the `tokenId` and make this method call the new one passing `token.uid`. What do you think?

File: index.js (hunk 6 above)
Comment (msbrogli) [reply index.js:238]: You're right. I think the `api-docs.js` is enough. Maybe we can add a docstring like "For the docs, see api-docs.js". What do you think?

**Review by msbrogli — APPROVED** [review]

File: index.js (hunk 4 above)
Comment (pedroferreira1) [reply index.js:206]: Created the issue. When we create the new method we can do exactly what you said.

File: index.js (hunk 5 above)
Comment (pedroferreira1) [reply index.js:225]: Resolved in the other comment

File: index.js (hunk 6 above)
Comment (pedroferreira1) [reply index.js:274]: I've added a short description of each endpoint and the reference to the docs

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**